Mathlib/NumberTheory/Padics/PadicIntegers.lean

```diff
@@ -85,26 +85,10 @@ theorem mem_subring_iff {x : ℚ_[p]} : x ∈ subring p ↔ ‖x‖ ≤ 1 := Iff
 
 variable {p}
 
-/-- Addition on `ℤ_[p]` is inherited from `ℚ_[p]`. -/
-instance : Add ℤ_[p] := (by infer_instance : Add (subring p))
-
-/-- Multiplication on `ℤ_[p]` is inherited from `ℚ_[p]`. -/
-instance : Mul ℤ_[p] := (by infer_instance : Mul (subring p))
-
-/-- Negation on `ℤ_[p]` is inherited from `ℚ_[p]`. -/
-instance : Neg ℤ_[p] := (by infer_instance : Neg (subring p))
-
-/-- Subtraction on `ℤ_[p]` is inherited from `ℚ_[p]`. -/
-instance : Sub ℤ_[p] := (by infer_instance : Sub (subring p))
-
-/-- Zero on `ℤ_[p]` is inherited from `ℚ_[p]`. -/
-instance : Zero ℤ_[p] := (by infer_instance : Zero (subring p))
```

Comment:
It wouldn't surprised me if there is a performance benefit to keeping these; possibly worth reverting from this PR to benchmark separately.

Reply:
From the benchmark results below, it seems that there is rather a performance benefit to getting rid of them.

Reply:
@eric-wieser Since the benchmark test establishes that this PR in fact causes a substantial net speed-up rather than the slow-down that you apparently feared, may I suggest we proceed?

Reply:
bors d=@mattrobball

I agree this looks fine in this case, but let's get a second opinion to confirm whether this is a rational fear of mine.